src/adagrams.js

```diff
@@ -1,15 +1,206 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const oringinalPool = [
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "A",
+    "B",
+    "B",
+    "C",
+    "C",
+    "D",
+    "D",
+    "D",
+    "D",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "E",
+    "F",
+    "F",
+    "G",
+    "G",
+    "G",
+    "H",
+    "H",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "I",
+    "J",
+    "K",
+    "L",
+    "L",
+    "L",
+    "L",
+    "M",
+    "M",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "N",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "O",
+    "P",
+    "P",
+    "Q",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "R",
+    "S",
+    "S",
+    "S",
+    "S",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "T",
+    "U",
+    "U",
+    "U",
+    "U",
+    "V",
+    "V",
+    "W",
+    "W",
+    "X",
+    "Y",
+    "Y",
+    "Z",
+  ];
+
+  let hand = [];
+  let gamePool = oringinalPool.slice();
+
+  while (hand.length < 10) {
+    let letter = gamePool[Math.floor(Math.random() * gamePool.length)];
+
+    hand.push(letter);
+
+    let index = gamePool.indexOf(letter);
+    if (index > -1) {
+      gamePool.splice(index, 1);
+    }
+  }
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let word = input.toUpperCase();
+  let copy = lettersInHand.slice();
+  let result = true;
+
+  for (let letter of word) {
+    if (copy.includes(letter)) {
+      let index = copy.indexOf(letter);
+      if (index > -1) {
+        copy.splice(index, 1);
+      }
+    } else {
+      result = false;
```

Comment:
Rather than saving `false` in a variable, you can just return `false` here. That way you can also break out of the for loop early. One line 134 you would just return `true`.